### ISSUE TITLE
Abbreviation tooltips

### ISSUE THREAD

**Issue by funderburkjim** [sanskrit-lexicon/MD#11]
This issue devoted to initial creation of a file from the 'LIST OF ABBREVIATIONS' provided in the front matter.
This was discussed [here](https://github.com/sanskrit-lexicon/csl-orig/issues/695).

![image](https://github.com/sanskrit-lexicon/MD/assets/6393033/3935c765-76d0-4637-bfa4-bc8281e28b3c)

**Comment by funderburkjim** [comment]
@AnnaRybakovaT 

Hi!

Instructions for first step prepared for you [here](https://github.com/sanskrit-lexicon/MD/tree/master/mdissues/issue11),

Let me know when you see this note.

**Comment by Andhrabharati** [comment]
@funderburkjim 

You guys need to keep [this also](https://github.com/sanskrit-lexicon/MD/issues/8#issuecomment-1435719277) in mind, to add at the end of the abbr. markup.

**Comment by Andhrabharati** [comment]
And then if required, I could finally step-in to identify the **unlisted** abbr.s present in the text, as I did an all other works.

**Comment by funderburkjim** [comment]
The particular task of this issue is to prepare an abbreviation tooltip file.  

A next step would be to add markup `<ab>X</ab>` in md.txt.  

Within that next step, (or possibly as a second next step), we can consider the 'asterisk'-related markup.

**Comment by Andhrabharati** [comment]
@funderburkjim 

Here are the listed abbr.s in the MD print--
[md-abbr.txt](https://github.com/sanskrit-lexicon/MD/files/12522516/md-abbr.txt)

As you might be somewhat free from pwk abbr. work now (till Thomas comes back to you), I am posting this file for your perusal and further action.
[I would just like to state here that I got quite many (~60) other abbr.s from the MD text.]

**Comment by Andhrabharati** [comment]
It may be interesting to note that MD has employed both regular cap. **N.** and the small cap. **ɴ.** as abbr.s (the ɴ. being used 400+ times in the text, while the N. is present 4200+ times).

**Comment by funderburkjim** [comment]
@Andhrabharati   acknowledged.  You are right - I'll be holding off changes to pwk until Thomas finishes.
I guess Anna is not available now.  Agree that abbreviation markup for MD is needed.

**Comment by Andhrabharati** [comment]
If you're willing to use it, I can post my MD file, with many corrections (I just don't want to list them) incorporated.

It is something like the GRA file (from me) that you've used recently.

**Comment by funderburkjim** [comment]
## abbrev1
`<ab>` markup applied, based on the list of abbreviations shown in first comment of this issue.
Working directory: https://github.com/sanskrit-lexicon/MD/tree/master/mdissues/issue11

* [temp_md_1.zip](https://github.com/sanskrit-lexicon/MD/files/12651938/temp_md_1.zip) the version of md with markup.  Displays based on this version are installed at Cologne.
* abbrev0 directory - preliminary work with the digitization provided by @Andhrabharati .
* abbrev1 directory: application of markup
  * [ab_count_1.txt](https://github.com/sanskrit-lexicon/MD/blob/master/mdissues/issue11/abbrev1/ab_count_1.txt) shows counts of the global abbreviations, along with tooltip text.
 * [ab_count_local_1.txt](https://github.com/sanskrit-lexicon/MD/blob/master/mdissues/issue11/abbrev1/ab_count_local_1.txt) shows the (few) local abbreviations.
 
@Andhrabharati  -  What do you think should be done next?  Are there differences between temp_md_1 and your version that you think the cdsl version should implement?

**Comment by Andhrabharati** [comment]
> Are there differences between temp_md_1 and your version that you think the cdsl version should implement?

Yes @funderburkjim, there are hundreds of types of changes (corrections), ranging from Sanskrit spellings (and/or accents) [sometimes even in headwords], English spellings, Greek spellings, wrong tags (italic, bold and sanskrit), hyphens, brace matching, … … …

And most important of them all is "decoding" the numeral marking ¤X¤ into various types, that I had mentioned earlier in a [response](https://github.com/sanskrit-lexicon/MD/issues/8#issuecomment-1434016963) to [your query](https://github.com/sanskrit-lexicon/MD/issues/8#issuecomment-1433915514).

Next comes the 'relocation' of the homonym numbers that you had inserted recently, to their 'proper' position as per MD print and intention!! 

You may look at various addl. tags that I had used--
`<ab></ab>`
`<bot></bot>`
`<cl></cl>`
`<fr></fr>`
`<gk></gk>`
`<hom></hom>`
`<lang></lang>`
`<lat></lat>`
`<lex></lex>`
`<ls></ls>`
`<pe></pe>`
`<zoo></zoo>`

Even if you would like to limit to abbr. markings, there are quite many yet to do, see for e.g. my extracted lists--
[MD ab_local.txt](https://github.com/sanskrit-lexicon/MD/files/12654848/MD.ab_local.txt)
[MD ab_global.txt](https://github.com/sanskrit-lexicon/MD/files/12654902/MD.ab_global.txt)
[And of course, there are many count differences as well between your version and AB version.]

**Comment by funderburkjim** [comment]
@Andhrabharati   Would you upload your version?

**Comment by Andhrabharati** [comment]
Here is the file for your study/reference, @funderburkjim--
[md_AB_v1.zip](https://github.com/sanskrit-lexicon/MD/files/12664446/md_AB_v1.zip)
- I had removed the italic markers around the ab-tags etc. for better readability, with a thought that they could be rendered in italics while displaying them.
- You may note that the '🞄' could be replaced by a line-break, to get closer (but not equal) to the cdsl version in terms of line count.

`---------------------------------------`
And here is the file, with a kind of semantic line-breaks for the sub-HWs inside the entries [they are not always the composite words formed from the main HW, but most of the times 'siblings' containing the first portion of the main entry!!]--
[md_AB_v2.zip](https://github.com/sanskrit-lexicon/MD/files/12664527/md_AB_v2.zip)
[This is just done as a trial, not as a full (complete) work.]

**Comment by gasyoun** [comment]
> Sanskrit spellings (and/or accents) [sometimes even in headwords]

headwords is what I value the most @Andhrabharati

**Comment by AnnaRybakovaT** [comment]
> Instructions for first step prepared for you [here](https://github.com/sanskrit-lexicon/MD/tree/master/mdissues/issue11),
> 
> Let me know when you see this note

Dear Jim and dear all,
Glad to see you after some months break.

During summer time you have mentioned me in some topics (BHS Issue 4 and PWK Issue 95). Please let me know from what I should start now?

Regardind this current issue I can't clone the directory ( I a bit confussed - is it : 
github.com/sanskrit-lexicon/MD/mdissues/issue11    or
github.com/sanskrit-lexicon/MD/tree/master/mdissues/issue11

in any case I had messages like:
```
fatal: repository 'https://github.com/sanskrit-lexicon/MD/tree/master/mdissues/issue11/' not found
---
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

**Comment by drdhaval2785** [comment]
I think you will have to do 

git clone https://github.com/sanskrit-lexicon/MD.git

**Comment by Andhrabharati** [comment]
@AnnaRybakovaT,

Jim has already done what he wanted you to do (as a first step) reg. the MD abbr.s; so you need not bother about the same now.

However, @funderburkjim  **is yet to take up further changes** based on [my posted file](https://github.com/sanskrit-lexicon/MD/issues/11#issuecomment-1726081122).

And the pwk-95 has nothing more to do; it has gone into many major changes in the later days.

As such, I think, you may now see if BHS-4 issue interests you, as Jim has suggested.
But probably Jim might wish to put you on some other task; so, **let's wait for his response.**

**Comment by AnnaRybakovaT** [comment]
> As such, I think, you may now see if BHS-4 issue interests you, as Jim has suggested.
> But probably Jim might wish to put you on some other task; so, **let's wait for his response.**

Great! Of course I would like to work with BHS. So I am waiting for the decision.

**Comment by funderburkjim** [comment]
@Andhrabharati  -- thank you for taking an interest here.  I would prefer for you to work with @AnnaRybakovaT  on either  this MD topic or the BHS topic (or both)
(https://github.com/sanskrit-lexicon/BHS/issues/4)  according to what your mutual interest suggests.

My part would then be limited to helping integrate your work into the active displays for MD, BHS.

If these tasks are completed while Anna is available,  then we might consider involving Anna in an AP90 task -- developing an  MW-style display  so that the AP90 nominal compounds would be accessible directly.

**Comment by Andhrabharati** [comment]
@funderburkjim 

I think, there is nothing more that Anna or I could do in this MD issue; it is ONLY you that can take-up further work based on [my file(s) posted above](https://github.com/sanskrit-lexicon/MD/issues/11#issuecomment-1726081122).

@AnnaRybakovaT 

Do you think you can proceed based on what Jim had suggested at BHS-4 (https://github.com/sanskrit-lexicon/BHS/issues/4), or need anything more?

**Comment by Andhrabharati** [comment]
On a 2nd thought [re-looking at Jim's posting], **apart from AP90**, I think my [md_AB_v2.zip](https://github.com/sanskrit-lexicon/MD/files/12664527/md_AB_v2.zip) file can be worked upon, to make this MD another work having sub-HWs 'accessible' to online search-queries, **after MW**.

And, my BEN file (already posted long ago) also can be a similar candidate.

@AnnaRybakovaT 
do you think you could take-up this piece of work, to make the 'full' HWs from the 'partial' sub-HWs, by appropriately filling up the (presumed) beginnings?
[Just browsing through my above file (look for `<div/>` tags) might give you some ideas!!]

**Comment by AnnaRybakovaT** [comment]
> Do you think you can proceed based on what Jim had suggested at BHS-4 ([sanskrit-lexicon/BHS#4](https://github.com/sanskrit-lexicon/BHS/issues/4)), or need anything more?

As I see - I have to work only with the file tagcount_ls.txt
In general I understood the task. Shall I make a copy of this file to make my changes in this copy?

**Comment by AnnaRybakovaT** [comment]
> 'full' HWs from the 'partial' sub-HWs, by appropriately filling up the (presumed) beginnings?

I am so sorry but I need more explanations regarding this task. First of all could you kindly show me exsamples in the file - what is:
-  'full' HWs
-  partial' sub-HWs

**Comment by Andhrabharati** [comment]
> As I see - I have to work only with the file tagcount_ls.txt
> In general I understood the task. Shall I make a copy of this file to make my changes in this copy?

That's correct; pl. go ahead.

**Comment by Andhrabharati** [comment]
> > 'full' HWs from the 'partial' sub-HWs, by appropriately filling up the (presumed) beginnings?
> 
> I am so sorry but I need more explanations regarding this task. First of all could you kindly show me exsamples in the file - what is:
> 
>     * 'full' HWs
> 
>     * partial' sub-HWs

@AnnaRybakovaT 
We shall come back to this MD task, after the above BHS work is done.

**Comment by Andhrabharati** [comment]
@funderburkjim 

Is it OK if we take-up the MD sub-HWs work before the AP90 (that you suggested)?

**Comment by funderburkjim** [comment]
> Is it OK if we take-up the MD sub-HWs

Yes - I will put work on your versions of MD on my (nearby) TODO list.

**Comment by funderburkjim** [comment]
## comments on version ab.v1
* The two points you mention ( 🞄  and the absence of italic markup {%%} around abbreviations)  can be handled in make_xml.py.  
* ab.v1 adds several tags (or changes tags, notably from `<ab>` to `<lex>`).
  * two of these tags (`<pe>` and `<cl>`) are new (not appearing in other dictionary markup).  
    This can be handled by revising one.dtd in csl-pywork.
      * note: minor typo (`<pe>X<pe>` corrected, 6 instances.

Here is a table summarizing the differences in tag markup

| tag | AB_v1 count | cdsl count |
| ----| ------------| ---------- |
|\<H> | 42 | 43 |
|\<ab-LOCAL> | 32 | 179 |
|\<ab> | 46925 | 100344 |
|\<bot> | 155 | 0 |
|\<cl> | 993 | 0 |
|\<fr> | 1 | 0 |
|\<gk> | 10 | 0 |
|\<hom> | 1363 | 920 |
|\<lang> | 102 | 0 |
|\<lang-LOCAL> | 0 | 9 |
|\<lbinfo-LOCAL>| 0 |  12342 |
|\<lat> | 3 | 0 |
|\<lex> | 56122 | 0 |
|\<ls> | 58 | 0 |
|\<pe> | 314 | 0 |
|\<zoo> | 15 | 0 |


@Andhrabharati   I think these differences account for most of the differences between the prior (cdsl) MD and the v1 version. 
*  Are there any other systematic differences in v1 that the above discussion misses?
*  In particular, is there any 'alternate headword'  markup  that remains to be done  in v1 ?

**Comment by Andhrabharati** [comment]
I don't think there are other tags etc.; but there sure would be spelling changes in my file wrt to cdsl version (as I had mentione previously). **And, there are no alt. HWs, as I recall in MD.**

If you're going to use my file as-is, then nothing more to 'study' in my file!!
[BTW, I thought you'd be taking up the MD after our (Anna and me) finishing the sub-HWs task as well.]

I remember using the `<pe>` tag in PWG and `<cl>` tag in MW in my recent working.
[However, both these are not yet posted out!]

**Comment by funderburkjim** [comment]
I'll go ahead and install v1, making changes in csl-pywork as mentioned.  Thanks for comment reminding me about  spelling corrections you made.

The 'v2'  (sub-hws)  task that you are working on  needs to be in another issue, which I'll open soon.   Our target should be like 'mw' -- Since we don't have any other 'sub-hw' dictionaries,  we need to think carefully how to proceed.  I'll elaborate on this when I open that other issue.

**Comment by funderburkjim** [comment]
## corr. tooltip
Happen to notice in v1 `<ab>corr.</ab>` (30 instances).  We need a tooltip.  'correlative' ?
(In MD's published list, there is 'cor.' correlative)
Do you recall any other items like corr.  (marked as `<ab>` (global), but with no tooltip in MD's list?

**Comment by funderburkjim** [comment]
AB.v1 (with a few minor changes) now installed as cdsl version.
[temp_md_ab_1pe.zip](https://github.com/sanskrit-lexicon/MD/files/13760229/temp_md_ab_1pe.zip) has the changes, which I think you should incorporate in further versions.

Work done in [abv1 directory](https://github.com/sanskrit-lexicon/MD/tree/master/mdissues/issue11/abv1).

Many new items added to the tooltips.  See this commit [([0469541](https://github.com/sanskrit-lexicon/csl-pywork/commit/046954138c1c492b2655a4b2e8688c77ef9cce9e)).
There are still 3 `<disp>??</disp>` with completely unresolved tooltips.
And several others with a ? in the tooltip where I was uncertain.

**Comment by Andhrabharati** [comment]
@funderburkjim 

>  `<cl> tag -- class of verb.  <cl>X</cl>  X is a roman-numeral`

>  `<cl>V.</cl> -> <cl>ᴠ.</cl>  class 5 (33 - to avoid conflict with <ab>V.</ab> Vedict 2470 instances`
>  `<cl>V.</ab> is class 5 root`
>   `Change to use Unicode U+1d20  Latin Letter Small Capital V`
>   `<cl>ᴠ.</ab>`

I would suggest to go for unicode Roman numerals (U+216x), throughout for the dhAtu class-numbers; as only 1-10 such cases [Ⅰ, Ⅱ, Ⅲ, Ⅳ, Ⅴ, Ⅵ, Ⅶ, Ⅷ, Ⅸ, Ⅹ] are required, we'd have no issues.
[**Using a small capital letter ᴠ (U+1d20) [as above] is not a proper choice**, as all other class numbers are in normal-size capital letters.]

Here is the updated file (hoping that you'd have no issue in agreeing to my proposal)--
[md_AB_v1.zip](https://github.com/sanskrit-lexicon/MD/files/13762163/md_AB_v1.zip)

**Comment by Andhrabharati** [comment]
> There are still 3 `<disp>??</disp>` with completely unresolved tooltips.

Are these the ones having the superscript numbers ¹ and ² ?

They stand for 'rare' (or singular) occurrences in the whole 'text' being referred.

See what MD says in his Preface (p. ⅸ),--

![image](https://github.com/sanskrit-lexicon/MD/assets/75209130/a36ad11d-1ce8-4b10-81a7-b589227d01c7)

 Also this snippet reminds you of the pending work that I had mentioned [above](https://github.com/sanskrit-lexicon/MD/issues/11#issuecomment-1673624909), to which you had also [responded](https://github.com/sanskrit-lexicon/MD/issues/11#issuecomment-1673660186) that **it would be taken up next**.

As I had already pointed this out (as above) to you, I **deliberately** did not mark it thus in my later working.

**Comment by funderburkjim** [comment]
## roman numeral revision.

@Andhrabharati  accepted your revision re roman numerals. It is now installed.  One correction:
```
At<L>19659<pc>358-2<k1>sfj
<ab>A.</ab> -> <lex>Ā.</lex>
```

Related revisions to mdab_input.txt in csl-pywork,  md-meta2 in csl-orig. 
For details, see the commits above or the mdissus/issue11/abv1 readme.txt.

Please note the ¹ and ²  tooltips in [mdab_input.txt](https://github.com/sanskrit-lexicon/csl-pywork/blob/master/v02/distinctfiles/md/pywork/mdab/mdab_input.txt). I translated these as 'one instance' or 'two instances' in the tooltips.

The three unknown abbrevs show as `<disp>??</disp>` in mdab_input.txt.

**Comment by funderburkjim** [comment]
> Also this snippet reminds you of the pending work that I had mentioned https://github.com/sanskrit-lexicon/MD/issues/11#issuecomment-1673624909, to which you had also https://github.com/sanskrit-lexicon/MD/issues/11#issuecomment-1673660186 that it would be taken up next.
As I had already pointed this out (as above) to you, I deliberately did not mark it thus in my later working.

@Andhrabharati  I prefer to omit further work on this (e.g. marking '*' as abbreviation, with associated tooltip).  Feel free to add such markup in a future version  (and DOCUMENT what you do).

**Comment by Andhrabharati** [comment]
> The three unknown abbrevs show as `<disp>??</disp>` in mdab_input.txt.

Filled these, and also corrected a few other abbr.s--
[mdab_input_AB.txt](https://github.com/sanskrit-lexicon/MD/files/13768806/mdab_input_AB.txt)

Here is my updated file--
[md_AB_v1.zip](https://github.com/sanskrit-lexicon/MD/files/13768757/md_AB_v1.zip)

And, just noticed that I did not change the 𝑃. (Purāṇa) occurrences inside the text file [**it is rendered as a normal letter in the print, being within the italic string(s)!**], though the abbr. list is having it thus. This shall be done while on v.2 (sub-HWs) work.

**Comment by funderburkjim** [comment]
@Andhrabharati  --  In your file [md_AB_v1.zip] https://github.com/sanskrit-lexicon/MD/files/13768757/md_AB_v1.zip from prior comment.

I see only 1 difference (under #upa at line 18596).  Is this what you intended?

**Comment by Andhrabharati** [comment]
Yes, while looking for **nl.** I've noticed the **vb. px.** here and thought it should've been **vbl. px.** (the upasarga)!

**Comment by funderburkjim** [comment]
## revisions installed.

Yes, I agree with the 'vbl.' change under 'upa', added a print change note in csl-corrections
Thanks for changes to mdab_input.  I made a couple of changes to your changes, 
  as noted in [abv1/readme.txt](https://github.com/sanskrit-lexicon/MD/tree/master/mdissues/issue11/abv1) at 
*12-26-2023  AB rev to mdab_input.txt*. 
 Or you can see via the commits above.  I'm fairly sure that, in mdab_input.txt, 'sts.'  is 'sometimes'  (even though 'st.' is 'stem'!).  I also used both 'absolute' (according to md print abbreviations) and 'absolutive' for 'abs.'

**Comment by Andhrabharati** [comment]
> I'm fairly sure that, in mdab_input.txt, 'sts.' is 'sometimes' (even though 'st.' is 'stem'!).

Yes-- you're correct, @funderburkjim ; I did not pay proper attention to the file content.

`----------------------------------`

> * In particular, is there any 'alternate headword'  markup  that remains to be done  in v1 ?



> **And, there are no alt. HWs, as I recall in MD.**

I was wrong here as well!!
Seen now, that the text has quite many alt. HW candidates; but these all need to be 'marked' yet [like in the GRA and pwk].
Do you think **this part** could be done **now**, or along with the sub-HWs task sometime **later**?

[I also noticed that I had missed quite many nuances in the text earlier. Too bad of me, that I did not put my mind properly in the MD work.]

**Comment by gasyoun** [comment]
> quite many alt. HW candidates

Interesting to know if any unique ones, as compared to other dictinaries, @Andhrabharati

**Comment by Andhrabharati** [comment]
I recall MD being helpful in resolving an issue or two while on MW two years back; no other work had those words!!

And yes @gasyoun, I think we might get few interesting entries from MD, if fully worked upon.

And I have noticed many entries in VCP, which are not anywhere else!!

**Comment by gasyoun** [comment]
> noticed many entries in VCP, which are not anywhere else

of utmost interest such your remarks. Would love to call you tomorrow and talk about what is your vision on future on Sanskrit dictionaries, what is done, what is important and what will remain to be done for generations ahead. @drdhaval2785 @funderburkjim @AnnaRybakovaT how about a call on 4th of January 20:00 Moscow time?

**Comment by drdhaval2785** [comment]
I will be able to spend time from 20:00 to 22:00 Indian Standard Time. 20:00 Moscow time will be too late in night for me.

**Comment by gasyoun** [comment]
> I will be able to spend time from 20:00 to 22:00 Indian Standard Time. 20:00 Moscow time will be too late in night for me.

I'm ready for a call 20 Indian Standard Time as well. Wrote in our sanskrit-lexicon Skype group.
